aquadoggo/src/network/service.rs

```diff
@@ -31,42 +29,47 @@ pub async fn network_service(
     // Subscribe to communication bus
     let mut _rx = tx.subscribe();
 
-    // Read the network configuration parameters from the application context
-    let network_config = context.config.network.clone();
-
     // Load the network key pair and peer ID
     let key_pair =
         NetworkConfiguration::load_or_generate_key_pair(context.config.base_path.clone())?;
-    let peer_id = PeerId::from(key_pair.public());
-    info!("Network service peer ID: {peer_id}");
-
-    let quic_config = quic::Config::new(&key_pair);
-    // Create QUIC transport (provides transport, security and multiplexing in a single protocol)
-    let quic_transport = quic::tokio::Transport::new(quic_config)
-        // Perform conversion to a StreamMuxerBox (QUIC handles multiplexing)
-        .map(|(p, c), _| (p, StreamMuxerBox::new(c)))
-        .boxed();
-    let quic_multiaddr =
-        format!("/ip4/0.0.0.0/udp/{}/quic-v1", network_config.quic_port).parse()?;
 
-    // Instantiate the custom network behaviour with default configuration
-    // and the libp2p peer ID
-    let behaviour = Behaviour::new(&network_config, peer_id, key_pair)?;
+    // Read the network configuration parameters from the application context
+    let network_config = context.config.network.clone();
+
+    // Build the network swarm and retrieve the local peer ID
+    let (mut swarm, local_peer_id) = swarm::build_swarm(&network_config, key_pair).await?;
 
-    // Initialise a swarm with QUIC transports, our composed network behaviour
-    // and the default configuration parameters
-    let mut swarm = SwarmBuilder::with_tokio_executor(quic_transport, behaviour, peer_id)
-        .connection_limits(network_config.connection_limits())
-        // This method expects a NonZeroU8 as input, hence the try_into conversion
-        .dial_concurrency_factor(network_config.dial_concurrency_factor.try_into()?)
-        .per_connection_event_buffer_size(network_config.per_connection_event_buffer_size)
-        .notify_handler_buffer_size(network_config.notify_handler_buffer_size.try_into()?)
-        .build();
+    // Define the QUIC multiaddress on which the swarm will listen for connections
+    let quic_multiaddr =
+        format!("/ip4/0.0.0.0/udp/{}/quic-v1", network_config.quic_port).parse()?;
 
     // Listen for incoming connection requests over the QUIC transport
     swarm.listen_on(quic_multiaddr)?;
 
-    // Dial each peer identified by the multi-address provided via `--remote-node-addresses` if given
+    let mut external_circuit_addr = None;
+
+    // Construct circuit relay addresses and listen on relayed address
```

Comment:
Sooo, this is the dedicated "relay message listener"?

Reply:
That's correct. It's our external relayed address (includes the address and peer ID of the relay server).